Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,141 @@ References:
+            | lookup RegistryType on $left.OldValueType == $right.TypeCode | project-rename RegistryPreviousValueType = TypeName
+            | project TimeGenerated, Computer, EventID, EventType, ActorUsername, ActorDomainName, ActorUserId, ActorSessionId, ActingProcessName, ActingProcessId, RegistryKey, _ResourceId, RegistryValueData, RegistryPreviousKey, RegistryPreviousValue,RegistryPreviousValueData, RegistryValue, RegistryValueType, RegistryPreviousValueType, Type
+        ), (
+            SecurityEvent
+            | where EventID == 4663
+                and ObjectType == "Key"
+            | lookup EventTypeLookup on AccessMask
+            | invoke ASIM_ParseSecurityEvents()
+            | project TimeGenerated, Computer, EventID, EventType, ActorUsername, ActorDomainName, ActorUserId, ActorSessionId, ActingProcessName, ActingProcessId, RegistryKey, _ResourceId, Type
+        ), (
+            SecurityEvent
+            | where EventID == 4657
+            | invoke ASIM_ParseSecurityEvents()
+            | extend EventOriginalSubType = OperationType
+                , RegistryValue = ObjectValueName
+            | lookup RegistryAction on EventOriginalSubType
+            | extend RegistryValueData = iff (EventOriginalSubType == "%%1906", OldValue, NewValue)
+                , RegistryPreviousKey = iff (EventOriginalSubType == "%%1905", RegistryKey, "")
+                , RegistryPreviousValue = iff (EventOriginalSubType == "%%1905", RegistryValue, "")
+                , RegistryPreviousValueData = iff (EventOriginalSubType == "%%1905", OldValue, "")
+            | lookup RegistryType on $left.NewValueType == $right.TypeCode | project-rename RegistryValueType = TypeName
+            | lookup RegistryType on $left.OldValueType == $right.TypeCode | project-rename RegistryPreviousValueType = TypeName
+            | project TimeGenerated, Computer, EventID, EventType, ActorUsername, ActorDomainName, ActorUserId, ActorSessionId, ActingProcessName, ActingProcessId, RegistryKey, _ResourceId, RegistryValueData, RegistryPreviousKey, RegistryPreviousValue,RegistryPreviousValueData, RegistryValueType, RegistryPreviousValueType, Type
+        )
+        | extend ActorUserType = ASIM_GetWindowsUserType(ActorUsername, ActorUserId)
+            , ActorUsernameType = ASIM_GetAccountType(ActorUserId)
+        | project-rename DvcFQDN = Computer
+        | extend User = ActorUsername
+            , UserId = ActorUserId
+            , Process = ActingProcessName
+            , Dvc = DvcFQDN
+            , EventStartTime = TimeGenerated
+            , EventEndTime = TimeGenerated
+            , EventOriginalType = tostring(EventID)
+        | extend EventSchemaVersion = "0.1" 
+            , EventSchema = "RegistryEvent"
+            , EventCount = toint(1)
+            , EventResult = "Success"
+            , EventVendor = "Microsoft"
+            , EventProduct = "Security Events" 
+            , DvcOs = "Windows"
+            , ActorUserIdType = "SID"
```

Comment:
Also need to set ActorUserSid as an alias to ActorUserId

Reply:
Check